vcd/datasource_vcd_vm_group.go

```diff
@@ -0,0 +1,68 @@
+package vcd
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+)
+
+// datasourceVcdVmGroup Defines the data source for a VM Group, used to create VM Placement Policies.
+func datasourceVcdVmGroup() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdVmGroupRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the VM Group to fetch",
+			},
+			"provider_vdc_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the Provider VDC to which the VM Group to fetch belongs",
+			},
+			"cluster_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the vSphere cluster associated to this VM Group",
+			},
+			"named_vm_group_id": {
```

Comment:
How does the `named_vm_group_id` differ from `vmGroup.VmGroup.ID`?

Reply:
As far as I could experiment, `namedVmGroupId` is **not** unique, where `vmGroupId` is.
I had an example where I created two VM Groups in vSphere with same name, one in a cluster and the other in another cluster. Turns out that their `namedVmGroupId` is the same, while the other attributes differ.

To reach uniqueness, one must use `<namedVmGroupId, pVdcId>` pairs, which is the combination UI uses (as for some reason they use named VM Group IDs and not regular IDs).